frontend/src/pages/ForgotPasswordPage.tsx

```diff
@@ -0,0 +1,251 @@
+import React, { useState, useEffect, FormEvent, useMemo } from 'react';
+import { Alert } from 'react-bootstrap';
+import { Trans, useTranslation } from 'react-i18next';
+import { Link } from 'react-router-dom';
+import classNames from 'classnames';
+import { ModalContent } from '../components/modals/modalparts/ModalContent';
+import { ModalHeader } from '../components/modals/modalparts/ModalHeader';
+import { Footer } from '../components/Footer';
+import { paths } from '../routers/paths';
+import { Input, errorState } from '../components/forms/FormField';
+import { LoadingSpinner } from '../components/LoadingSpinner';
+import { ReactComponent as MailIcon } from '../icons/mail_icon.svg';
+import { api } from '../api/api';
+import css from './ForgotPassword.module.scss';
+
+const classes = classNames.bind(css);
+
+export const ForgotPasswordPage = () => {
+  const { t } = useTranslation();
+  const defaultDisableTime = 30; // Seconds
+
+  const [linkSent, setLinkSent] = useState(false);
+  const [email, setEmail] = useState('');
+  const [sendDisabled, setSendDisabled] = useState(false);
+  const [sendDisableTime, setSendDisableTime] = useState(
+    defaultDisableTime / 2,
+  );
+  const [resendDisabled, setResendDisabled] = useState(false);
+  const [resendDisableTime, setResendDisableTime] = useState(
+    defaultDisableTime,
+  );
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  let timeoutArray = useMemo(() => {
+    const array: NodeJS.Timeout[] = [];
+    return array;
+  }, []);
+
+  let intervalArray = useMemo(() => {
+    const array: NodeJS.Timeout[] = [];
+    return array;
+  }, []);
+
+  // Cleanup timers after unmounting
+  useEffect(
+    () => () => {
+      timeoutArray.forEach((timeout) => {
+        clearTimeout(timeout);
+      });
+      intervalArray.forEach((interval) => {
+        clearInterval(interval);
+      });
+    },
+    [timeoutArray, intervalArray],
+  );
```

Comment:
Do you mean generally or in this specific scenario? My reasoning for the two different timers here is that I made them to serve different purposes.

The first timeout is for setting timer to the "Send link" -button on the sendLinkView. It starts when user requests a link on that first view. The reason for this timeout is that you can't spam emails by sending link -> change address -> send link etc. If user is using this page correctly, they should never even know about this timeout, that's why it's 15 seconds instead of the 30 which is on the second view as 30 seconds could have effect on normal use as well eg. accidentally entering wrong email and realizing it in 10 seconds and then having to wait another 20. This is just a small protection against malicious use.

The second timeout is for the "Resend link" -button on the linkSentView. It starts when users requests a new link (after the first email didn't supposedly arrive). The reason for this being different timer than the sendLink one is that it didn't look as good and smooth (to me) if the "Didn’t receive the email? Resend link" was already disabled when entering the page the first time.  

And that's why both of these timers could be active at the same time since they get set from different actions and I had to clean them both in the cleanup

Reply:
Maybe I'm thinking about these timers way too complicatedly and the pages could just share the timers 😬

Reply:
> the pages could just share the timers

This is what I was thinking. It would also keep the time consistent when the user navigates to change email during the 30 s timer.

> This is just a small protection against malicious use.

Any enforcement of the send rate must be done on the server side anyway, but it's good that the frontend indicates that the user should not spam the button if they didn't receive the email in a heartbeat.